core/zip/CMakeLists.txt

```diff
@@ -9,14 +9,23 @@ Set(ZipOldHeaders
   ${CMAKE_CURRENT_SOURCE_DIR}/inc/RZip.h
 )
 
+Set(ZipNewHeaders
+  ${CMAKE_CURRENT_SOURCE_DIR}/inc/zlib.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/inc/zconf.h
+)
+
 Set(ZipOldSource
   ${CMAKE_CURRENT_SOURCE_DIR}/src/ZDeflate.c
   ${CMAKE_CURRENT_SOURCE_DIR}/src/ZInflate.c
 )
 
 # Depending on the system either the old and the new
 # or only the old compression is build
-set(ZLIB_HEADERS ${ZipOldHeaders})
+if(MSVC AND builtin_zlib)
```

Comment:
I wonder if we don't need to always add these to avoid things like [this issue](https://sft.its.cern.ch/jira/browse/ROOT-8839). If the headers are not there, there will certainly be a mismatch with the one in the system when users `#include <zlib.h>`. That is, maybe the `MSVC` part is not needed and we should just `set(ZLIB_HEADERS ${ZipNewHeaders} ${ZipOldHeaders})` if `builtin_zlib` is `ON`.

Reply:
It's what I originally did, but since it was supposed to be working on other platforms, I decided to add the `if(MSVC)`

Reply:
Ah, I see. I can merge with it in place and test removing it later, then. However, I think removing the `if (MSVC)` is at least in part something we need for fixing the issue above, so if you'd like to remove and test in the PR, go for it. Otherwise, let me know and I'll merge.

Reply:
So you can remove it if you wish (or I can remove it tomorrow) I know it will work on Windows anyway...